### DISCUSSION TITLE
Unit testing new attribute, constructor with parameters not being found

### DISCUSSION THREAD

**Discussion by peteraritchie** [riok/mapperly#466] [Q&A]
I thought I'd attempt to add the ability to add explicit enum value mapping via a `MapEnumValueAttribute` attribute.  I've got the `Attribute` class with a constructor and I'm trying to unit test.  I'm extending an existing unit test that uses the `MapEnumAttribute` attribute...  But I'm getting an error in the test that the `MapEnumValueAttribute` constructor can't be found (there's only one constructor, public, with two parameters).  `MapEnumAttribute` has a single public constructor with one parameter and that's working.  It's like the new class isn't being included in the syntax tree of the test.  `MapEnumValueAttribute` is adjacent to `MapEnumAttribute` and in the same project.

I've added the details to PublicAPI.Unshipped.txt (and also tried PublicAPI.Shipped.txt).

Has something like this been encountered before, or is there something I've missed?

**Comment by TimothyMakkison** [comment]
Are you able to build and run the sample/unit tests before making these changes? Sometimes restarting your ide helps.

> Has something like this been encountered before, or is there something I've missed?

I've had a similar error that appeared in Visual Studio build order, perhaps it's related: 
`Generator 'MapperGenerator' failed to generate source. Exception was of type 'InvalidOperationException' with message 'Type Riok.Mapperly.Abstractions.EnumMappingStrategy, Riok.Mapperly.Abstractions not found`

It randomly started happening a week ago, I assumed it was caused by a VS update because it doesn't happen with rider

**Reply by peteraritchie** [reply]
I didn't see that one, but was getting other errors that ended up causing the issue.

**Comment by peteraritchie** [comment]
There were some compile errors in the source code text.  Adding public to the types and the method also helped.
